donut/modules/bodfeedback/helpers.py

```diff
@@ -0,0 +1,255 @@
+
+
+def get_id(uuid):
+    '''
+    Returns the complaint_id associated with a uuid
+    or false if the uuid is not found
+    '''
+    query = 'SELECT complaint_id FROM bod_complaint_info WHERE uuid = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, str(uuid))
+        if not cursor.rowcount:
+            return False
+    return cursor.fetchone()['complaint_id']
+
+
+def get_messages(complaint_id):
+    '''
+    Returns timestamps, posters, messages, and message_id's on this complaint
+    in ascending order of timestamp
+    '''
+    query = """
+    SELECT time, poster, message, message_id FROM bod_complaint_messages WHERE complaint_id = %s ORDER BY time
+    """
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchall()
+    if not (res and 'message_id' in res[0]):
+        return None
+    return res
+
+
+def get_summary(complaint_id):
+    '''
+    Returns a dict with the following fields: subject, status
+    '''
+    query = 'SELECT subject, status FROM bod_complaint_info WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchone()
+    return res
+
+
+def get_subject(complaint_id):
+    '''
+    Returns the suject or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['subject'] if res else None
+
+
+def get_status(complaint_id):
+    '''
+    Returns the status of a post or None if complaint_id is invalid
+    '''
+    res = get_summary(complaint_id)
+    return res['status'] if res else None
+
+
+def mark_read(complaint_id):
+    '''
+    Sets the status of this complaint to 'read'
+    returns False if complaint_id is invalid
+    '''
+    query = """
+    UPDATE bod_complaint_info SET status = 'read' WHERE complaint_id = %s
+    """
+    if get_status(complaint_id) is None:
+        return False
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+
+
+def mark_unread(complaint_id):
+    '''
+    Sets the status of this complaint to 'new_msg'
+    returns False if complaint_id is invalid
+    '''
+    query = """
+    UPDATE bod_complaint_info SET status = 'new_msg' WHERE complaint_id = %s
+    """
+    if get_status(complaint_id) is None:
+        return False
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        # Not sure why this doesn't autocommit
+        cursor.execute("COMMIT;")
+
+
+def get_emails(complaint_id):
+    '''
+    Returns a list of subscribed emails for this complaint (which may be empty)
+    or an empty list if complaint_id is invalid
+    '''
+    query = 'SELECT email FROM bod_complaint_emails WHERE complaint_id = %s'
+    with flask.g.pymysql_db.cursor() as cursor:
+        cursor.execute(query, complaint_id)
+        res = cursor.fetchall()
+    return [row['email'] for row in res]
```

Comment:
same potential issue if `res` is not set

Reply:
Unless an exception occurs in the `with` block, `res` should always be set